### PR TITLE
feat/triangle_loader

### PR DESCRIPTION
**Pull request by fraiss-c**

### CODE REVIEW

File: coverage.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #
 # MIT License
 #
-# Copyright (c) 2017-2018 Thibault Martinez and Simon Ninon
+# Copyright (c) 2017-2018 Charles Fraïssé and Michaël Leclerc
```
Comment (MickaToast) [inline coverage.sh]: MIT requires giving credit to author, this this file is copied from their repositories, we have to use their header

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,7 @@ set(SOURCE_FILES    src/Camera/Camera.cc
                     src/Dispatcher/Dispatcher.cc
                     src/Engine/Color.cc
                     src/Engine/Engine.cc
+                    src/Mesh/Mesh.cc
```
Comment (MickaToast) [inline CMakeLists.txt]: No test file for mesh?

File: src/Engine/Engine.cc [hunk 3]
```diff
@@ -21,15 +21,52 @@ OR OTHER DEALINGS IN THE SOFTWARE. */
...
     }
 
     Engine::~Engine() {
     }
 
     Color Engine::raytrace(const rt::Vector2<int> &pixel) {
-        Color color(0xff0000ff); // Red color
-        //TODO: logic of raytracing a point will be here
-        //TODO: _camera.generateRay(pixel);
+        Color color(0x00000000);
```
Comment (MickaToast) [inline src/Engine/Engine.cc]: Color() will create a color with 0x00000000 component, use it instead

File: src/Engine/Engine.cc [hunk 4]
```diff
@@ -21,15 +21,52 @@ OR OTHER DEALINGS IN THE SOFTWARE. */
...
+        float min = -1;
+        Intersection inter;
+        Ray ray = _camera.GenerateRay(pixel);
+        for (int i = 0; i < _triangles.size(); i++) { //TODO: KDTree
```
Comment (MickaToast) [inline src/Engine/Engine.cc]: Prefer using ++i in for-loop condition

File: src/Engine/Engine.cc [hunk 5]
```diff
@@ -21,15 +21,52 @@ OR OTHER DEALINGS IN THE SOFTWARE. */
...
+            if (inter.Intersect) {
+                Vector3<float> dist = inter.Point - _camera.GetPos();
+                if (min == -1 || dist.Norm() < min) {
+                    min = dist.Norm();
+                    float angle = ray.Direction.Angle(_triangles[i].GetNormal());
+                    if (angle > 90.f) {
+                        angle = 180.f - angle;
+                    }
+                    float coef = (-1.f / 90.f) * angle + 1.f;
+                    color.SetRedComponent(_triangles[i].GetMaterial().Ka.X * 255 * coef);
```
Comment (MickaToast) [inline src/Engine/Engine.cc]: `255 * coef` can be set in memory

File: src/Engine/Engine_test.cc [hunk 6]
```diff
@@ -24,8 +24,14 @@ OR OTHER DEALINGS IN THE SOFTWARE. */
 namespace rt {
```
Comment (MickaToast) [inline src/Engine/Engine_test.cc]: Can you provide more tests for this scene? A ray that fully intersect the object and a ray that have an antialiased pixel

File: src/Vector/Vector2_test.cc [hunk 7]
```diff
@@ -88,4 +89,15 @@ namespace rt {
...
         EXPECT_EQ(v * 6, Vector2<int>(6, 12));
     }
+
+    TEST(Vector2_int, operatorDivision) {
+        Vector2<int> v = Vector2<int>(4, 2);
+
+        EXPECT_EQ(v / 2, Vector2<int>(2, 1));
+    }
+
+    TEST(Vector2_int, Display) {
```
Comment (MickaToast) [inline src/Vector/Vector2_test.cc]: Test not testing (no assert)

File: CMakeLists.txt (hunk 2 above)
Comment (fraiss-c) [reply CMakeLists.txt]: Tested in Triangle as it is an abstract class

**Pull request merged**